### ISSUE TITLE
DINO training loss 

### ISSUE THREAD

**Issue by Faisal-Hajari** [lightly-ai/lightly#1000]
Hello, 
I'm training resnet18 on a custom dataset. it's been running for some time with a batch size of  325 (the max my GPU can handle). the thing is the loss is flat and it's not getting better or worse. 

is this behavior normal ? and if so how do you decide on stopping the training ?

**Comment by Faisal-Hajari** [comment]
I tried a lot of tweeks from adding/removing warmups, changing the optim and the learning rate, different heads..etc.

**Comment by Faisal-Hajari** [comment]
>There is some error in the parameters or configuration of the network, thus the training loss converges too quickly.
I'm using (for the most part the default params) and the model seems to learn good embedding (I tested by loading the student, and only use it to embed a sample of the dataset and ran a random forest on top of the embed) and there is a difference in performance depending on which epoch the student has been pulled from. 

>The network works, but is fully trained out on your given dataset. This can happen e.g. if you have too little or similar images.
I do think this is more likely (I trained on 38K retinal images) that do look similar and have a lot of black-background. 

so my question is: 
1) do we only relay on the DINO loss to pick which model is better trained ? in other words if we use check point system, we should probably use the one with the lowest loss. or is there metrics for self-supervised embedding.

**Comment by MalteEbner** [comment]
> Do we only rely on the DINO loss to pick which model is better trained ? in other words if we use check point system, we should probably use the one with the lowest loss. or is there metrics for self-supervised embedding.

Yes, usually the DINO loss is used. If the loss does not change, then your model parameters probably don't change anymore and your network has converged. There is nothing unusual about that. If the embedding are good enough for your external evaluation, then this is fine and the SSL has done its job.

Alternatively to using the checkpoint with the lowest training loss, you can also use another evaluation method, e.g. run a random forest on top of the embedding every epoch and see how the random forest performance on a validation set changes. For more details on how to implement such a custom callback, see https://github.com/lightly-ai/lightly/issues/977
